src/documents/models.py

```diff
@@ -267,17 +277,90 @@ def __str__(self):
         return str(created)
 
     @property
-    def source_path(self):
+    def source_filename(self):
+        if self.filename is None:
+            self.filename = self.generate_source_filename()
+
+        return self.filename
+
+    @staticmethod
+    def many_to_dictionary(field):
+        # Converts ManyToManyField to dictionary by assuming, that field
+        # entries contain an _ or - which will be used as a delimeter
+        mydictionary = dict()
+
+        for t in field.all():
+            # Find delimeter
+            delimeter = t.name.find('_')
+
+            if delimeter is -1:
+                delimeter = t.name.find('-')
+
+            if delimeter is -1:
+                continue
+
+            key = t.name[:delimeter]
+            value = t.name[delimeter+1:]
+
+            mydictionary[slugify(key)] = slugify(value)
+
+        return mydictionary
+
+    @staticmethod
+    def many_to_list(field):
+        # Converts ManyToManyField to list
+        mylist = list()
+
+        for t in field.all():
+            mylist.append(slugify(t))
```

Comment:
Does `slugify(t)` do the right thing, or should this be `slugify(t.name)`? I.e. does `str(t)` return the name of the tag? (Maybe we should be explicit here rather than rely on implicit behaviour that might change.)

Reply:
Fixed (although the result for the moment is obviously the same, but being more explicit might prevent problems in the future).